dask/array/core.py

```diff
@@ -1111,6 +1111,10 @@ def _set_chunks(self, chunks):
     chunks = property(_get_chunks, _set_chunks, "chunks property")
 
     def __len__(self):
+        if len(self.chunks) == 0:
+            raise ValueError("Can not get length if chunks are empty. "
+                             "Please initalize with chunks or use"
+                             " rechunk method to set if needed.")
```

Comment:
We should error similar to numpy zero-dim arrays here:

```python
In [11]: import numpy as np

In [12]: x = np.array(1)

In [13]: len(x)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-13-a7f4a5366567> in <module>()
----> 1 len(x)

TypeError: len() of unsized object
```

Something like:

```
raise TypeError("len() of unsized object")
```

would be more accurate.

Reply:
That makes more sense also - I was actually wondering what the use case for having chunkless-arrays were to begin with - as I'd think this would normally get caught in the `__new__` where `normalize_chunks` is called.